src/Transports.AspNetCore/GraphQLHttpMiddleware.cs

```diff
@@ -126,22 +114,25 @@ public async Task InvokeAsync(HttpContext context)
                     var request = gqlBatchRequest[i];
 
                     var stopwatch = ValueStopwatch.StartNew();
-                    var result = await executer.ExecuteAsync(
-                        request.OperationName,
-                        request.Query,
-                        request.GetInputs(),
-                        userContext,
-                        token).ConfigureAwait(false);
+                    var result = await ExecuteRequestAsync(request, userContext, executer, cancellationToken);
 
                     await RequestExecutedAsync(new GraphQLRequestExecutionResult(gqlRequest, result, stopwatch.Elapsed, i));
 
                     executionResults[i] = result;
                 }
 
-                await WriteResponseAsync(context, writer, executionResults).ConfigureAwait(false);
+                await WriteResponseAsync(context, writer, cancellationToken, executionResults).ConfigureAwait(false);
             }
         }
 
+        private static async Task<ExecutionResult> ExecuteRequestAsync(GraphQLRequest gqlRequest, IDictionary<string, object> userContext, IGraphQLExecuter<TSchema> executer, CancellationToken token)
```

Comment:
no need in async/await

Reply:
I'd thought that too, but the `ConfigureAwait(false)` stuffs it up.

Reply:
No need in `ConfigureAwait(false)` since this method has nothing to do with async:

```c#
private static Task<ExecutionResult> ExecuteRequestAsync(GraphQLRequest gqlRequest, IDictionary<string, object> userContext, IGraphQLExecuter<TSchema> executer, CancellationToken token)
            => executer.ExecuteAsync(
                gqlRequest.OperationName,
                gqlRequest.Query,
                gqlRequest.GetInputs(),
                userContext,
                token);
```

Reply:
Good point, will remove.